src/app/input/input.directive.ts

```diff
@@ -79,7 +79,7 @@ export class MzInputDirective extends HandlePropChanges implements OnInit, OnDes
     const labelElement = document.createElement('label');
     labelElement.setAttribute('for', this.id);
 
-    this.renderer.invokeElementMethod(this.inputContainerElement, 'append', [labelElement]);
+    this.renderer.invokeElementMethod(this.inputElement, 'after', [labelElement]);
```

Comment:
This should be fixed everywhere we create the label element (find all `createLabelElement` methods)